src/platform-aws.c

```diff
@@ -737,3 +737,103 @@ int platform_config_endpoint(struct fi_info *info, struct fid_ep* endpoint) {
 exit:
 	return ret;
 }
+
+static int get_rail_vf_idx(struct fi_info *info)
+{
+	char dev_name[224], guid_file[256], guid[20];
+	char *ptr, *endptr;
+	int vf_idx;
+	FILE *fp;
+
+	/* get ib device name and read GUID from sysfs */
+	strncpy(dev_name, info->domain_attr->name, sizeof(dev_name)-1);
```

Comment:
Ick.  Why not `info->nic->device_attr->name`, and then you can skip the whole dev_name entirely.